mlflow/gateway/config.py

```diff
@@ -271,18 +271,24 @@ def to_route(self) -> "Route":
         return Route(
             name=self.name,
             route_type=self.route_type,
-            model=ModelInfo(
+            model=RouteModelInfo(
                 name=self.model.name,
                 provider=self.model.provider,
```

Comment:
Should we check this 'self.model.provider' is 'str' instead of 'Provider'?

Reply:
`Provider` is like a str-line enum, so no validation should be required :)